packages/sitecore-jss/src/layout-service.ts

```diff
@@ -0,0 +1,100 @@
+import { AxiosDataFetcher } from './data-fetcher';
+import { LayoutServiceData } from './dataModels';
+import * as dataApi from './dataApi';
+import { HttpJsonFetcher } from './httpClientInterface';
+import { IncomingMessage, ServerResponse } from 'http';
+
+export type LayoutServiceConfig = {
+  apiHost: string;
+  apiKey: string;
+  siteName: string;
+  /**
+   * Enables/disables analytics tracking for the Layout Service invocation (default is true).
+   * More than likely, this would be set to false for SSG/hybrid implementations, and the
+   * JSS tracker would instead be used on the client-side: {@link https://jss.sitecore.com/docs/fundamentals/services/tracking}
+   * @default true
+   */
+  tracking?: boolean;
+  /**
+   * Custom data fetcher. @see HttpJsonFetcher<T> type for implementation details/notes.
+   * By default used @see AxiosDataFetcher
+   * @param {string} url The URL to request; may include query string
+   * @param {any} [data] Data to POST with the request.
+   * @param {IncomingMessage} [req] Request instance
+   * @param {ServerResponse} [res] Response instance
+   */
+  dataFetcher?: HttpJsonFetcher<LayoutServiceData>;
```

Comment:
The `HttpJsonFetcher` doesn't accept the `req`/`res` parameters directly, but I think it is important we support a custom dataFetcher that accepts them. Probably need to switch this up to more of a factory pattern (like the `getDefaultFetcher` below) - which accepts `req`/`res` and returns `HttpJsonFetcher`.